### PR TITLE
PAYARA-3240 Cleanup sonar warnings for nucleus/cluster/admin

### PR DESCRIPTION
**Pull request by Cousjava**

### CODE REVIEW

File: nucleus/cluster/admin/src/main/java/com/sun/enterprise/v3/admin/cluster/CreateInstanceCommand.java [hunk 1]
```diff
@@ -552,17 +549,7 @@ private String makeCommandHuman(List<String> command) {
...
-            return false;
-
-        if(theNode.getType().equals("SSH"))
-            return false;
-
-        if(theNode.getType().equals("DCOM"))
-            return false;
-
-        return true;
+        return !(theNode.isLocal() || theNode.getType().equals("SSH") || theNode.getType().equals("DCOM"));
```
Comment (MarkWareham) [inline nucleus/cluster/admin/src/main/java/com/sun/enterprise/v3/admin/cluster/CreateInstanceCommand.java]: I preferred the old way :-P

File: nucleus/cluster/admin/src/main/java/com/sun/enterprise/v3/admin/cluster/CreateNodeImplicitCommand.java [hunk 2]
```diff
@@ -53,7 +53,6 @@
 
 
 import org.jvnet.hk2.annotations.Service;
-import org.jvnet.hk2.component.*;
```
Comment (MarkWareham) [inline nucleus/cluster/admin/src/main/java/com/sun/enterprise/v3/admin/cluster/CreateNodeImplicitCommand.java]: Portions copyright missing

File: nucleus/cluster/admin/src/main/java/com/sun/enterprise/v3/admin/cluster/CreateNodeSshCommand.java [hunk 3]
```diff
@@ -47,11 +47,9 @@
 import org.glassfish.api.I18n;
 import org.glassfish.api.Param;
 import org.glassfish.api.admin.*;
-import javax.inject.Inject;
 
 
 import org.jvnet.hk2.annotations.Service;
-import org.jvnet.hk2.component.*;
```
Comment (MarkWareham) [inline nucleus/cluster/admin/src/main/java/com/sun/enterprise/v3/admin/cluster/CreateNodeSshCommand.java]: Portions copyright missing

File: nucleus/cluster/admin/src/main/java/com/sun/enterprise/v3/admin/cluster/NodeAgentConfigUpgrade.java [hunk 4]
```diff
@@ -140,7 +140,6 @@ public Object run(Domain d) throws PropertyVetoException, TransactionFailure {
                         s.setNodeAgentRef(null);
                     }
                     //remove the node-agent element
```
Comment (MarkWareham) [inline nucleus/cluster/admin/src/main/java/com/sun/enterprise/v3/admin/cluster/NodeAgentConfigUpgrade.java]: comment should be removed too.

File: nucleus/cluster/admin/src/main/java/com/sun/enterprise/v3/admin/cluster/NodeUtils.java [hunk 5]
```diff
@@ -502,28 +498,23 @@ private void validateSSHConnection(ParameterMap map) throws
...
+                }
+            
+        } catch (IOException e) {
             String m1 = e.getMessage();
             String m2 = "";
             Throwable e2 = e.getCause();
             if (e2 != null) {
                 m2 = e2.getMessage();
             }
-            if (e instanceof FileNotFoundException) {
```
Comment (MarkWareham) [inline nucleus/cluster/admin/src/main/java/com/sun/enterprise/v3/admin/cluster/NodeUtils.java]: wow

File: nucleus/cluster/admin/src/main/java/com/sun/enterprise/v3/admin/cluster/RestartInstanceCommand.java [hunk 6]
```diff
@@ -313,10 +313,7 @@ private boolean isInstanceRestartable() throws InstanceNotRunningException {
         }
 
         String val = rac.findPropertyInReport("restartable");
-        if (val != null && val.equals("false")) {
-            return false;
-        }
-        return true;
+        return Boolean.parseBoolean(val);
```
Comment (MarkWareham) [inline nucleus/cluster/admin/src/main/java/com/sun/enterprise/v3/admin/cluster/RestartInstanceCommand.java]: Needs reverting, or it's a bug. Carefully inspect this again. If val == null ?

File: nucleus/cluster/admin/src/main/java/com/sun/enterprise/v3/admin/cluster/ValidateNodeCommand.java [hunk 7]
```diff
@@ -152,26 +159,25 @@ public void execute(AdminCommandContext context) {
...
-            logger.fine(Strings.get(
-                            "Node {0} is valid. Updating if needed", name));
+            logger.fine(Strings.get("Node {0} is valid. Updating if needed", name));
 
         // What is there in the node is valid. Now go update anything that
         // was not there.
         CommandInvocation ci = cr.getCommandInvocation("_update-node", report, context.getSubject());
         ParameterMap map = new ParameterMap();
         map.add("DEFAULT", name);
-        if (! excludeFromUpdate.contains("installdir"))
```
Comment (MarkWareham) [inline nucleus/cluster/admin/src/main/java/com/sun/enterprise/v3/admin/cluster/ValidateNodeCommand.java]: could do a addToMapIfNotExcludeFromUpdate(String key) method here.

**Comment by Cousjava** [comment]
Jenkins test please

File: nucleus/cluster/admin/src/main/java/com/sun/enterprise/v3/admin/cluster/BootstrapSecureAdminCommand.java [hunk 8]
```diff
@@ -68,8 +68,8 @@
 })
 public class BootstrapSecureAdminCommand implements AdminCommand, PostConstruct {
 
-    private final static String DOWNLOADED_FILE_MIME_TYPE = "application/octet-stream";
-    private final static String DOWNLOAD_DATA_REQUEST_NAME = "secure-admin";
+    private static final String DOWNLOADED_FILE_MIME_TYPE = "application/octet-stream";
```
Comment (MarkWareham) [inline nucleus/cluster/admin/src/main/java/com/sun/enterprise/v3/admin/cluster/BootstrapSecureAdminCommand.java]: Portions copyright missing

File: nucleus/cluster/admin/src/main/java/com/sun/enterprise/v3/admin/cluster/DeleteNodeSshCommand.java [hunk 9]
```diff
@@ -47,11 +47,8 @@
 import org.glassfish.api.admin.*;
 import org.glassfish.hk2.api.PerLookup;
 
-import javax.inject.Inject;
-
```
Comment (MarkWareham) [inline nucleus/cluster/admin/src/main/java/com/sun/enterprise/v3/admin/cluster/DeleteNodeSshCommand.java]: Portions copyright missing

File: nucleus/cluster/admin/src/main/java/com/sun/enterprise/v3/admin/cluster/ValidateNodeCommand.java (hunk 7 above)
Comment (MarkWareham) [reply nucleus/cluster/admin/src/main/java/com/sun/enterprise/v3/admin/cluster/ValidateNodeCommand.java]: Not quite what I meant. The idea was to remove the large amount of IFs

File: nucleus/cluster/admin/src/main/java/com/sun/enterprise/v3/admin/cluster/dcom/PingNodeDcomCommand.java [hunk 10]
```diff
@@ -44,7 +44,6 @@
 import com.sun.enterprise.v3.admin.cluster.NodeUtils;
 import com.sun.enterprise.v3.admin.cluster.PingNodeRemoteCommand;
 import org.glassfish.api.admin.*;
-import javax.inject.Inject;
```
Comment (MarkWareham) [inline nucleus/cluster/admin/src/main/java/com/sun/enterprise/v3/admin/cluster/dcom/PingNodeDcomCommand.java]: Missing portions copyright

File: nucleus/cluster/admin/src/main/java/com/sun/enterprise/v3/admin/cluster/CopyConfigCommand.java [hunk 11]
```diff
@@ -75,8 +75,7 @@
 })
 public final class CopyConfigCommand extends CopyConfig {
 
-    final private static LocalStringManagerImpl localStrings =
-        new LocalStringManagerImpl(CopyConfigCommand.class);
+    private static final LocalStringManagerImpl localStrings = new LocalStringManagerImpl(CopyConfigCommand.class);
```
Comment (MarkWareham) [inline nucleus/cluster/admin/src/main/java/com/sun/enterprise/v3/admin/cluster/CopyConfigCommand.java]: Missing portions copyright

**Comment by Cousjava** [comment]
Jenkins test please

**Comment by Cousjava** [comment]
Jenkins test please

File: nucleus/cluster/admin/src/main/java/com/sun/enterprise/v3/admin/cluster/ValidateDcom.java [hunk 12]
```diff
@@ -322,11 +324,10 @@ private void dumpStack(Throwable t) {
 
     private boolean testPort(int port, String description) {
         try {
-            Socket socket = new Socket();
-            socket.connect(new InetSocketAddress(host, port), 4000);
-            socket.close();
-            out.append(Strings.get("validate.dcom.connect",
-                    description, port, host)).append('\n');
+            try (Socket socket = new Socket()) {
```
Comment (svendiedrichsen) [inline nucleus/cluster/admin/src/main/java/com/sun/enterprise/v3/admin/cluster/ValidateDcom.java]: If you do this at the outer try the IOException will also be caught with the outer catch. Inner try-with-resources is obsolete.

File: nucleus/cluster/admin/src/main/java/com/sun/enterprise/v3/admin/cluster/ValidateNodeCommand.java [hunk 13]
```diff
@@ -152,33 +159,32 @@ public void execute(AdminCommandContext context) {
...
-            map.add("sshkeyfile", sshkeyfile);
+        addIfNotExcluded(INSTALLDIR, installdir, map);
+        addIfNotExcluded(NODEHOST, nodehost, map);
+        addIfNotExcluded(NODEDIR, nodedir, map);
+        addIfNotExcluded(SSHPORT, sshport, map);
+        addIfNotExcluded(SSHUSER, sshuser, map);
+        addIfNotExcluded(SSHKEYFILE, sshkeyfile, map);
 
         // Only update if there is something to do
         if ( map.size() > 1) {
```
Comment (svendiedrichsen) [inline nucleus/cluster/admin/src/main/java/com/sun/enterprise/v3/admin/cluster/ValidateNodeCommand.java]: !map.isEmpty()

Comment (svendiedrichsen) [inline nucleus/cluster/admin/src/main/java/com/sun/enterprise/v3/admin/cluster/ValidateNodeCommand.java]: !map.isEmpty()

**Pull request merged**